target/linux/pistachio/base-files/lib/netifd/proto/lowpan.sh

```diff
@@ -26,11 +27,10 @@ proto_lowpan_setup() {
         if [ ! -e /proc/device-tree/uccp@18480000/mac-address0 ]; then
                 logger "lowpan: Could not find wifi STA MAC address"
         else
-                local WIFI_STA_MAC=0x$(hexdump -n 6 -v -e '/1 "%02X"' /proc/device-tree/uccp@18480000/mac-address0)
-
-                # 6LOWPAN_MAC = WIFI_STA_MAC + 5
+                local PRIMARY_MAC_ADDR=$(get_primary_macaddr)
+                # 6LOWPAN_MAC = PRIMARY_MAC_ADDR + 5
                 local LOWPAN_OTP_OFFSET=0x5
-                local LOWPAN_MAC=$(printf "%012X" $((WIFI_STA_MAC + LOWPAN_OTP_OFFSET)))
+                local LOWPAN_MAC=$(printf "%012X" $((PRIMARY_MAC_ADDR + LOWPAN_OTP_OFFSET)))
```

Comment:
Should we keep the MAC related stuff in mac.sh and let the caller call get_lowpan_mac() ?
This way we can make the this script generic, as we will take the platform specific OTP layout stuff  out from lowpan.sh